modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/ViewScene.java

```diff
@@ -152,26 +164,46 @@ public void finishInputMethodComposition() {
         platformView.finishInputMethodComposition();
     }
 
-    @Override public String toString() {
-        View view = getPlatformView();
-        return (" scene: " + hashCode() + " @ (" + view.getWidth() + "," + view.getHeight() + ")");
+    @Override
+    public void processOverlayCSS() {
+        if (viewSceneOverlay != null) {
+            viewSceneOverlay.processCSS();
+        }
     }
 
-    void synchroniseOverlayWarning() {
-        try {
-            waitForSynchronization();
-            OverlayWarning warning = getWindowStage().getWarning();
-            if (warning == null) {
-                painter.setOverlayRoot(null);
-            } else {
-                painter.setOverlayRoot(NodeHelper.getPeer(warning));
-                warning.updateBounds();
-                NodeHelper.updatePeer(warning);
-            }
-        } finally {
-            releaseSynchronization(true);
-            entireSceneNeedsRepaint();
+    @Override
+    public void layoutOverlay() {
+        if (viewSceneOverlay != null) {
+            viewSceneOverlay.layout();
+        }
+    }
+
+    @Override
+    public void synchronizeOverlay() {
+        if (viewSceneOverlay != null) {
+            viewSceneOverlay.synchronize();
+        }
+    }
+
+    public void setViewSize(float width, float height) {
+        sceneListener.changedSize(width, height);
+
+        if (viewSceneOverlay != null) {
+            viewSceneOverlay.resize(width, height);
         }
     }
+
+    public void setOverlay(Parent root) {
+        overlayRoot = root;
+
+        if (viewSceneOverlay != null) {
+            viewSceneOverlay.setRoot(root);
+        }
+    }
+
+    @Override public String toString() {
+        View view = getPlatformView();
+        return (" scene: " + hashCode() + " @ (" + view.getWidth() + "," + view.getHeight() + ")");
```

Comment:
leading space in `toString()` might be a _bad idea_

suggestion (JSON-like):

`return "ViewScene" + hashCode() + "{width=" + view.getWidth() + ", height=" + view.getHeight() + "}";`

Reply:
Yep, but this is existing code. The GitHub diff viewer makes it seem like it was changed, but it isn't.